sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/GenericTypeParameterUnused.cs

```diff
@@ -122,17 +123,28 @@ public class WithLocalFunctions
     {
         public void Method()
         {
-            DoMoreStuff<int, string>("value");
+            void LocalFunctionNoTypeParameter() { }
 
-            T3 DoStuff<T, T1, T3>(params T3[] o) // Compliant - FN: local functions are not verified
+            void LocalFunctionUnusedParameter<T>() { } // Noncompliant
+
+            T LocalFunctionWithUsedTypeParameter<T>(T p) => p;
```

Comment:
from what I can see, all compliant tests have the type parameter as both return type and parameters type

it would be useful to have a test with only return
```
T LocalFunctionWithUsedTypeParameter<T>(object o) => (T)o;
```
and a test with only parameter
```
void LocalFunctionWithUsedTypeParameter<T>(T o) => o;
```

and also a test with many parameters and different combinations that might "break" the algorithm
```
static void StaticLocalFunctionWithUnusedTypeParameter<T1, T2, T3, T4, T5>(T1, T5, T5, T1) // Noncompliant - should mention 2 parameters which are not used : T2, T3 and T4

static void StaticLocalFunctionWithUnusedTypeParameter<T1, T2, T3, T4>(T4, T3, T2, T1)
```